minichain/chain.py

```
@@ -28,22 +30,41 @@ class Blockchain:
+        self._create_genesis_block(genesis_path)
 
-    def _create_genesis_block(self):
+    def _create_genesis_block(self, genesis_path):
         """
-        Creates the genesis block with a fixed hash.
+        Creates the genesis block and initializes state from config.
         """
+        config = {}
+        if os.path.exists(genesis_path):
+            try:
+                with open(genesis_path, "r") as f:
```

Ruff (0.15.14)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)